src/Illuminate/Container/ContextualBindingBuilder.php

```diff
@@ -66,4 +66,23 @@ public function give($implementation)
             $this->container->addContextualBinding($concrete, $this->needs, $implementation);
         }
     }
+
+    /**
+     * Define tagged services to be used as the implementation for the contextual binding.
+     *
+     * @param  string  $tag
+     * @return void
+     */
+    public function giveTagged($tag)
+    {
+        if (! $this->needs) {
+            $this->needs = $tag;
+        }
+
+        $this->give(function ($container) use ($tag) {
```

Comment:
`$container` can be typehinted with the container interface typehint for better IDE autocompletion.